desc/plotting.py

```diff
@@ -3764,6 +3785,141 @@ def plot_field_lines_real_space(
         return fig, ax
 
 
+def _plot_omnigenous_field(
+    field,
+    iota,
+    grid=None,
+    fill=False,
+    ncontours=30,
+    fieldlines=0,
+    ax=None,
+    return_data=False,
+    **kwargs,
+):
+    """Plot :math:`|B|` on a surface vs the Boozer poloidal and toroidal angles.
+
+    Parameters
+    ----------
+    field : OmnigenousField
+        Object from which to plot.
+    iota : float
+        Rotational transform.
+    rho : float, optional
+        Radial coordinate of flux surface.
+    fill : bool, optional
+        Whether the contours are filled, i.e. whether to use `contourf` or `contour`.
+    ncontours : int, optional
+        Number of contours to plot.
+    fieldlines : int, optional
+        Number of (linearly spaced) magnetic fieldlines to plot. Default is 0 (none).
+    ax : matplotlib AxesSubplot, optional
+        Axis to plot on.
+    return_data : bool
+        if True, return the data plotted as well as fig,ax
+    **kwargs : dict, optional
+        Specify properties of the figure, axis, and plot appearance e.g.::
+
+            plot_X(figsize=(4,6),cmap="plasma")
+
+        Valid keyword arguments are:
+
+        * ``figsize``: tuple of length 2, the size of the figure (to be passed to
+          matplotlib)
+        * ``cmap``: str, matplotlib colormap scheme to use, passed to ax.contourf
+        * ``levels``: int or array-like, passed to contourf
+        * ``title_fontsize``: integer, font size of the title
+        * ``xlabel_fontsize``: float, fontsize of the xlabel
+        * ``ylabel_fontsize``: float, fontsize of the ylabel
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        figure being plotted to
+    ax : matplotlib.axes.Axes or ndarray of Axes
+        axes being plotted to
+    plot_data : dict
+        dictionary of the data plotted, only returned if ``return_data=True``
+
+    Examples
+    --------
+    .. image:: ../../_static/images/plotting/plot_omnigenous_field.png
+
+    .. code-block:: python
+
+        from desc.plotting import plot_omnigenous_field
+        fig, ax = plot_omnigenous_field(field)
+
+    """
+    if grid is None:
+        grid = LinearGrid(rho=[1.0], theta=101, zeta=101, endpoint=False, NFP=field.NFP)
```

Comment:
a default is assigned in the plot_boozer_surface function, so this is never called? why not just have the defaults occur here?